apps/api-extractor/src/api/CompilerState.ts

```diff
@@ -67,6 +67,15 @@ export class CompilerState {
       );
     }
 
+    // Delete outDir and declarationDir to prevent TypeScript from redirecting self-package
+    // imports to source files. When these options are set, TypeScript's module resolution
+    // tries to map output .d.ts files back to their source .ts files to avoid analyzing
+    // build outputs during compilation. However, API Extractor specifically wants to analyze
+    // the .d.ts build artifacts, not the source files. Since API Extractor doesn't emit any
+    // files, these options are unnecessary and interfere with correct module resolution.
+    delete commandLine.options.outDir;
+    delete commandLine.options.declarationDir;
```

Comment:
Very insightful!  I've seen this problem before, but I didn't realize the import analysis was affected by `outDir` and `declarationDir`.